### PR TITLE
Fix duplicated consumers and connectivity issues

### PR DESCRIPTION
**Pull request by GPortas**
**Fix duplicated consumers and connectivity issues**
* * *

# What does this Pull Request do?
After doing end-to-end testing, we found some issues related to the service consumers. Sometimes more than one queue consumer was generated and this caused problems when listening to messages, in particular we found some messages that were not consumed and remained enqueued. This caused the process for some ingests to terminate at this point.

duplicated_consumers_1

The implementation of the service listener has been modified so that it is initialized at the same time as the Flask app. The new implementation is based on the listener model developed in DIMS, which has disconnection management and connection retries within the base listener itself.

The listener model is based on OOP and has been adapted to the existing code, which is more function-oriented. I have tried to impact as little as possible on the original design, focusing the changes only on what is necessary to solve the issues.

Service logging has also evolved, activating log file rotation, following the same approach as DIMS or HDHS.

# How should this be tested?

A description of what steps someone could take to:
* Execute tests through docker-compose-local.yml
* Send messages to the dims-data-ready queue and verify that only one consumer is active and that the messages are dequeued and the process flow works as expected.

# Test coverage
Yes/No: Are changes in this pull-request covered by:
- integration tests Yes

# Interested parties
@jessjass

### CODE REVIEW

File: app/mqresources/listener/drs_complete_queue_listener.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        self._logger.debug(
+            "Received message from DRS Queue. Message body: {}. Message id: {}".format(
+                str(message_body),
+                message_id
+            )
+        )
+        try:
+            # Send ingest status message to process queue
+            # TODO - this will have to get pull the actual URN once it is available
+            urn = "https://nrs-dev.lts.harvard.edu/URN-3:HUL.TEST:101113553"
```
Comment (awoods) [inline app/mqresources/listener/drs_complete_queue_listener.py:32]: @GPortas : can you please create a ticket in the backlog for replacing this hardcoded value?

File: app/mqresources/listener/drs_complete_queue_listener.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            mqutils.notify_ingest_status_process_message(
+                message_body["package_id"],
```
Comment (awoods) [inline app/mqresources/listener/drs_complete_queue_listener.py:34]: Would it be helpful to add the error-handling such as what was done in: https://github.com/harvard-lts/drs-import-management-service/commit/1ec4c6673dd4c911862e86e0a8afa4c5ed8d6f7e#diff-094615ad72ed85b45d5deaa6d831d6a379304033da1ef6a757bf3514a074e4abR37

**Review by awoods — COMMENTED** [review]
Looks good. 

Minor comment: It appears that your IDE automatically injects whitespace reformatting... which adds clutter to the PR. No need to change anything in this PR, but generally, it is helpful to limit the edits to functional changes.

File: app/mqresources/listener/drs_complete_queue_listener.py (hunk 2 above)
Comment (GPortas) [reply app/mqresources/listener/drs_complete_queue_listener.py:34]: Sure. This kind of improvement is what I refer to in the previous comment.

**Review by jessjass — APPROVED** [review]
Tested, looks good!

**Pull request merged**